03_python_fastapi_project/main.py

```
@@ -128,6 +161,179 @@ async def delete_product(product_id: int, db: AsyncSession = Depends(get_db)):
+            for item in cart.items
+        ],
+        "total_items": total_items,
+        "total_price": total_price
+    }
+
+
+@app.post("/cart/{session_id}/items")
+async def add_to_cart(
+    session_id: str, 
+    item_data: CartItemCreate, 
+    db: AsyncSession = Depends(get_db)
```

Ruff (0.13.3)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)